apps/ngrx/7-power-of-effect/src/app/app.effects.ts

```diff
@@ -0,0 +1,57 @@
+import { inject, Injectable } from '@angular/core';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { Actions, createEffect, ofType } from '@ngrx/effects';
+import { map, tap } from 'rxjs';
+import { schoolActions } from './school/school.actions';
+import { studentActions } from './student/store/student.actions';
+import { teacherActions } from './teacher/store/teacher.actions';
+
+const actionMap = new Map<string, string>();
+actionMap.set(schoolActions.addOneSchool.type, 'School added');
+actionMap.set(studentActions.addOneStudent.type, 'Student added');
+actionMap.set(teacherActions.addOneTeacher.type, 'Teacher added');
+
+@Injectable()
+export class AppEffects {
+  private actions$ = inject(Actions);
+  private snackbar = inject(MatSnackBar);
+
+  displayAlerts$ = createEffect(
+    () =>
+      this.actions$.pipe(
+        ofType(
+          schoolActions.addOneSchool,
+          studentActions.addOneStudent,
+          teacherActions.addOneTeacher,
+        ),
+        map((action) => {
+          switch (action.type) {
```

Comment:
I'm not a big fan of switch statement in effect. You should create separate effects for each use cases

Reply:
I just pushed two commits:

1. One to split the effect into four effects (1 for each of the three notification types, and one to do the actual alert).
2. Converted to functional effects (just for practice).